web/api/web_api_v1.c

```diff
@@ -615,12 +615,73 @@ inline int web_client_api_request_v1_registry(RRDHOST *host, struct web_client *
     }
 }
 
+inline void web_client_api_request_v1_info_summary_alarm_statuses(const RRDHOST *host, BUFFER *wb) {
+    int alarm_normal = 0, alarm_warn = 0, alarm_crit = 0;
+    RRDCALC *rc;
+    rrdhost_rdlock(host);
+    for(rc = host->alarms; rc ; rc = rc->next) {
+        if(unlikely(!rc->rrdset || !rc->rrdset->last_collected_time.tv_sec))
+            continue;
+
+        switch(rc->status) {
+            case RRDCALC_STATUS_WARNING:
+                alarm_warn++;
+                break;
+            case RRDCALC_STATUS_CRITICAL:
+                alarm_crit++;
+                break;
+            default:
+                alarm_normal++;
+        }
+    }
+    rrdhost_unlock(host);
+    buffer_sprintf(wb, "\t\t\"normal\": %d,\n", alarm_normal);
+    buffer_sprintf(wb, "\t\t\"warning\": %d,\n", alarm_warn);
+    buffer_sprintf(wb, "\t\t\"critical\": %d\n", alarm_crit);
+}
+
+inline void web_client_api_request_v1_info_mirrored_hosts(const RRDHOST *localhost, BUFFER *wb) {
+    RRDHOST *rc;
```

Comment:
hm... just realized this:

The API can be called as:

- http://your.netdata.ip:19999/api/v1/info
- http://your.netdata.ip:19999/host/NAME/api/v1/info

In the later case, the function parameter `localhost` will be a pointer to the host specified at the URL.

There is a also a global `RRDHOST *` called `localhost`, which you shadow here.

So, I suggest to name the function parameter `host`, remove variable `rc` and use `host` to iterate, starting at `localhost`, so that no matter which host you call the `info` method, the list of hosts will always be the same.

Reply:
@ktsaou I'm not clear. Do you mean like this?
```c
inline void web_client_api_request_v1_info_mirrored_hosts(const RRDHOST *host, BUFFER *wb) {
    RRDHOST *rc;
    int count = 0;
    rrd_rdlock();
    rrdhost_foreach_read(rc) {
        if(unlikely(rc == host)) continue;
        if(unlikely(count == 0)) {
            buffer_sprintf(wb, "\t\t\"%s\"", rc->hostname);
        } else {
            buffer_sprintf(wb, ",\n\t\t\"%s\"", rc->hostname);
        }
        count++;
    }
    if(count > 0)  buffer_strcat(wb, "\n");
    rrd_unlock();
}
```

Reply:
yes, but don't exclude the host given as parameter. So, give the whole list, every time.

Reply:
I've fixed it.